CONTRIBUTING.md

```diff
@@ -0,0 +1,609 @@
+  - [Step 7: Merge](#step-6-merge)
+- [Troubleshooting](#troubleshooting)
+- [Related Repositories](#related-repositories)
+- [Finding Contributions](#finding-contributions-to-work-on)
+- [Code of Conduct](#code-of-conduct)
+- [Security Issue Notifications](#security-issue-notifications)
+- [Licensing](#licensing)
+
+## Getting Started
+
+To get started with contributing to the AWS iOS SDK, first make sure you have the latest version of [Xcode](https://developer.apple.com/xcode/) installed. There are many different ways to build frameworks for iOS developers but we love to use [Cocoapods](https://guides.cocoapods.org/using/getting-started.html) and these instructions will include how to get started with Cocoapods so make sure that is installed as well if you'd like to follow along. You can install cocoapods by simply running:
+
+```bash
+sudo gem install cocoapods
+```
+
+Then make sure you fork the project and then clone your fork by running:
+
+```bash
+# change your GITHUBUSERNAME to your github username before running this command.
+git clone git@github.com:GITHUBUSERNAME/aws-sdk-ios.git
+```
+The project is broken into two major Xcode Projects:
+
+- Individual SDKs that target AWS services- AWSiOSSDKv2.xcodeproj located under / (root)
+- AuthSDKs (including AWSMobileClient) - AWSAuthSDK.xcodeproj located under /AWSAuthSDK
+
+ Example of commands to open the main SDK project that lives at the root of the repo  below:
+
+```bash
+cd aws-sdk-ios
+xed .
+```
+
+Conversely if you wanted to make a change to our Auth SDK's the initial workflow might be:
+
+```bash
+cd aws-sdk-ios
+cd AWSAuthSDK
+xed .
+```
+
+## Tools
+
+[Xcode](https://developer.apple.com/xcode/) and [Cocoapods](https://guides.cocoapods.org/using/getting-started.html) are used for all build and dependency management.
+
+## Integrated Frameworks
+
+Some widely used dependencies that have been copied into our project are:
+
+[Mantle](https://github.com/Mantle/Mantle) is what we use for deserialization and serialization and any modeling with JSON in our AWS iOS SDK.
+
+[SocketRocket](https://github.com/facebook/SocketRocket) is used for web socket connections.
+
+[Bolts-ObjC](https://github.com/facebookarchive/Bolts-ObjC) is used for "Tasks", which make organization of complex asynchronous code more manageable. A task is kind of like a JavaScript Promise but available for Objective C.
+
+### Integration Testing Setup
+
+Our integration tests are designed to execute on your development machine and talk to live AWS services. You may want to create these resources in a separate AWS account or at least tag them to not affect any resources you are using in production or confuse them. These tests are based on XCTest and require a json file with your credentials. If you are looking to run any of the integration tests in the main Xcode project, you will need to create a file named `credentials.json` located in AWSCoreTests> Resources>credentials.json and it should have the following format:
```

Comment:
`in a separate AWS account` -- might not be practical guidance for a non-AWS contributors. I believe you have to provide a credit card for billing when you sign-up a regular account. It's sort of an involved affair.

Reply:
well if they work for a company that uses AWS they may not want to do in their production AWS account meaning like directing them to a testing accoutn they have.

Reply:
but yes if you don't haev an AWS account you can't contribute unfortunately